File: DesignPatternsExercises/java/StrategyDesignPattern/Solution/CSVExtraction.java
class CSVExtraction implements Extraction {
    @Override
    public void extract() {
        // __________(2)__________;
        open();
        System.out.println("CSV Data type is getting extracted by separating row by commas...");
        // __________(3)__________;
        close();
    }

    @Override
    public void open() {
        System.out.println("Opening CSV file...");
    }

    @Override
    public void close() {
        System.out.println("Closing CSV file...");
    }
}

File: DesignPatternsExercises/ruby/strategy_design_pattern/exercise/csv_extraction.rb
require_relative 'extraction'

class CSVExtraction < Extraction
  def extract
    __________(2)__________
    puts 'CSV Data type is getting extracted by separating row by commas...'
    __________(3)__________
  end

  def open
    puts 'Opening CSV file...'
  end

  def close
    puts 'Closing CSV file...'
  end
end
